### PR TITLE
docs: Add instructions for custom certs for executors

### PR DESCRIPTION
**Pull request by sanderginn**
- Closes https://github.com/sourcegraph/sourcegraph/issues/51052

Adds instructions on how to load custom certificates for various deployment models of executors.

## Test plan
`sg run docsite`

### CODE REVIEW

File: doc/admin/executors/deploy_executors.md [hunk 1]
```diff
@@ -198,3 +198,76 @@ For Google Container Registry, [follow this guide](https://cloud.google.com/cont
...
+- mountPath: /etc/ssl/certs
+  name: custom-certs
+  readOnly: true
+```
+
+Next, apply the updated YAML manifests. Once the executors have rolled out, they should be picking up your custom certificates.
+
+### Adding certificates to a Kubernetes deployment using Helm
+
+Using custom certificates with a Helm deployment is not yet supported.
```
Comment (sanderginn) [inline doc/admin/executors/deploy_executors.md:263]: Is this worth mentioning or should it just be added once it actually is possible

**Comment by sanderginn** [comment]
Followup issue:
- https://github.com/sourcegraph/sourcegraph/issues/51312

File: doc/admin/executors/deploy_executors.md [hunk 2]
```diff
@@ -198,3 +198,76 @@ For Google Container Registry, [follow this guide](https://cloud.google.com/cont
...
+| `/etc/ssl/certs`                                    | SLES10/SLES11             |
+| `/etc/pki/tls/certs`                                | Fedora/RHEL               |   
+| `/system/etc/security/cacerts`                      | Android                   |
+
+If your environment makes use of custom certificates, you can add them to one of these locations in order for executors to pick them up.
+        
+### Adding certificates to a binary deployment
+> NOTE: see the [troubleshooting guide](./executors_troubleshooting.md#connecting-to-cloud-provider-executor-instances) for instructions on how to connect to cloud provider VMs.
+        
+After successfully [deploying binaries](./deploy_executors_binary.md), copy your certificates to `/etc/ssl/certs`. Next, run `executor run` on the VM in order to restart the executor service.
```
Comment (danieldides) [inline doc/admin/executors/deploy_executors.md:223]: A user may need to run `systemctl restart executor` if they're using systemd.

File: doc/admin/executors/deploy_executors.md (hunk 1 above)
Comment (danieldides) [reply doc/admin/executors/deploy_executors.md:263]: I think the Kubernetes instructions alone are sufficient. If you're running helm, you can just follow those anyway.

**Review by danieldides — APPROVED** [review]

File: doc/admin/executors/deploy_executors.md (hunk 1 above)
Comment (Piszmog) [reply doc/admin/executors/deploy_executors.md:263]: If the above K8s instructions work for Helm, maybe have a call out that the above could work for Helm.

**Review by Piszmog — COMMENTED** [review]
Thank you for this! It will help with a lot of churn we have seen lately.

**Pull request merged**